platform/storage/zephyr/nvs/src/mender-storage.c

```diff
@@ -181,8 +227,16 @@ mender_storage_set_deployment_data(char *deployment_data) {
 
     assert(NULL != deployment_data);
 
-    /* Write deployment data */
-    if (nvs_write(&mender_storage_nvs_handle, MENDER_STORAGE_NVS_DEPLOYMENT_DATA, deployment_data, strlen(deployment_data) + 1) < 0) {
+    size_t data_len = strlen(deployment_data) + 1;
+
+#ifdef CONFIG_MENDER_STORAGE_DEPLOYMENT_DATA_CRC
+    if (MENDER_OK != add_crc(&deployment_data, &data_len)) {
+        return MENDER_FAIL;
+    }
+#endif /* CONFIG_MENDER_STORAGE_DEPLOYMENT_DATA_CRC */
+
+    /* Write deployment data  */
+    if (nvs_write(&mender_storage_nvs_handle, MENDER_STORAGE_NVS_DEPLOYMENT_DATA, deployment_data, data_len) < 0) {
```

Comment:
Well, it would make the code simpler in a way by not having to do `realloc()` and point `uint32_t` to somewhere after string data, etc. But YMMV. The only functional difference is that by writing the CRC32 first, you immediately invalidate the old deployment data (if any) and so after a sudden reboot it will not be considered valid. However, I don't see any case where this would make a practical difference in our use case, TBH.

Reply:
By appending it to the end of the data, we could reuse the functions if we want to apply it to other data, that way we can avoid the need for a separate key per CRC

Reply:
True.